src/StatusQ/Components/StatusImageCropPanel.qml

```diff
@@ -117,15 +112,9 @@ Item {
         id: d
 
         function updateAspectRatio(newAR) {
-            // Keep width and adjust height
-            const eW = cropEditor.cropRect.width
-            const w = (eW <= 0) ? cropEditor.sourceSize.width : eW
-            const h = w/newAR
-            const c = (eW <= 0)
-                    ? Qt.point(cropEditor.sourceSize.width/2, cropEditor.sourceSize.height/2)
-                    : Qt.point(cropEditor.cropRect.x + w/2, cropEditor.cropRect.y + cropEditor.cropRect.height/2)
-            const nR = Qt.rect(c.x - w/2, c.y-h/2, w, h)
-            cropEditor.setCropRect(nR)
+            if(root.sourceSize.width === 0 || root.sourceSize.height === 0)
+                return
+            cropEditor.setCropRect(cropEditor.fillContentInWindow(root.sourceSize, Qt.size(1000, 1000/root.aspectRatio)))
```

Comment:
Probably it is better to store size as property and be able to overwrite default 1000 value